### ISSUE TITLE
`dzx compile` permisisons issues

### ISSUE THREAD

**Issue by andrewbrey** [c4spar/deno-dzx#4]
Say for example you have the following script, which requires read access to CWD, and you wanted to compile it with `dzx compile`:

```ts
/// <reference path="https://deno.land/x/dzx/types.d.ts" />

const pwd = await $o`pwd`;

console.log(`Your current working dir is ${pwd}`);
```

then, following the instructions in the readme, which specifies the following 

> dzx compile ./example.ts --allow-read --allow-env --allow-run

is how you compile your file into a standalone binary, you run:

```bash
dzx compile ./pwd.ts --allow-read
```

the `dzx compile` command will produce the following output:

![image](https://user-images.githubusercontent.com/34140052/130873903-64dd2f74-c0a5-4377-bce6-ecacccfba567.png)

It appears that, as of now, it is not possible to pass Deno runtime flags to the compilation, and also, it appears that attempting to do so will trigger an odd looking block in the compile command code 

https://github.com/c4spar/deno-dzx/blob/d7f40a64cc57eb95d03d400a8dfbccc0a9487039/src/cli/compile.ts#L32-L35

which _appears_ to be checking for if you're passing flags _before_ the script argument and swaps things around...honestly I'm not certain what the intent was here? Could it be checking if you're trying to read the script from stdin using the convention that a single `-` arg means "read from stdin"? That wasn't what I would expect given that the stdin case is handled in the next block, but I'm not certain :)

In summary, the compile flags are not being parsed correctly it seems and so, as of now, the inclusion of any flags causes the `dzx compile` command to think that the first flag you pass is the script you want to compile, and it throws up.
